### PR TITLE
IdanS scrapping capability

### PR DESCRIPTION
**Pull request by idan2468**
Initialization of scrapping functionality for the MoodleBooster
See issue #155 for details

### CODE REVIEW

File: content_scripts/storageLoader.js [hunk 1]
```diff
@@ -219,6 +219,46 @@ function listenForBackgroundMessages() {
...
+ *                                                   otherwise just one DOM element
+ */
+async function scrapeWebsiteDOM(url, cssSelector, all = false) {
+    let parser = new DOMParser()
+    let response = await fetch(url)
+    const contentType = response.headers.get('Content-Type')
+    const charsetStartIndex = contentType.lastIndexOf("charset=")
+    let htmlDoc
+    if (charsetStartIndex !== -1) {
+        const charSet = contentType.substring(charsetStartIndex + 8)
```
Comment (yanirelfasy) [inline content_scripts/storageLoader.js]: Why 8? maybe add a documentation  :)

File: content_scripts/storageLoader.js [hunk 2]
```diff
@@ -219,6 +219,46 @@ function listenForBackgroundMessages() {
...
+    return all ? htmlDoc.querySelectorAll(cssSelector) : htmlDoc.querySelector(cssSelector)
+}
+
+/**
+ *  Initial scrapping functionality from Syllabus to get in which semester is a course by Syllabus content
+ * @param courseId  The ID of the course to check
+ * @returns {Promise<string>} "a" for semester A and "b" for semester B
+ */
+async function getCourseSemester(courseId) {
+    if (courseId.length < 5) {
```
Comment (yanirelfasy) [inline content_scripts/storageLoader.js]: also here, why are we checking that? why number 5?

File: content_scripts/storageLoader.js (hunk 1 above)
Comment (idan2468) [reply content_scripts/storageLoader.js]: 8 as the string length "charset=",
I'll change it to be more clear and add some comment if it's not clear enough

File: content_scripts/storageLoader.js (hunk 2 above)
Comment (idan2468) [reply content_scripts/storageLoader.js]: I'll add a comment and maybe constant, but for now, I'll explain here too.
If the course id is with less than 5 chars then in the URL for the course's syllabus they add a prefix of 0s so the id will be with 5 chars.

**Review by yanirelfasy — APPROVED** [review]
Looks good, we need to add documentation for consts. 
for now it's good if we want to merge :)

**Pull request merged**